jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/ContractionHierarchyAlgorithm.java

```diff
@@ -0,0 +1,1154 @@
+                           ContractionVertex<V> vertexToIgnore) {
+
+        for (ContractionEdge<E> edge : graph.outgoingEdgesOf(vertex)) {
+            ContractionVertex<V> successor = graph.getEdgeTarget(edge);
+
+            double edgeWeight = graph.getEdgeWeight(edge);
+
+            if (edgeWeight < 0) {
+                throw new IllegalArgumentException("Negative edge weight not allowed");
+            }
+
+            if (successor.equals(vertexToIgnore) ||
+                    (verticesDataArray[successor.vertexId] != null && verticesDataArray[successor.vertexId].isIndependent)) {
+                // skip independent vertices because they do not belong to overlay graph
+                continue;
+            }
+
+            double updatedDistance = vertexDistance + edgeWeight;
+
+            updateDistance(successor, updatedDistance, heap, distanceMap);
+        }
+    }
+
+    /**
+     * Updates distance for {@code vertex} in the {@code heap} if needed.
+     *
+     * @param vertex      vertex
+     * @param distance    updated distance
+     * @param heap        search priority queue
+     * @param distanceMap vertices distances
+     */
+    private void updateDistance(ContractionVertex<V> vertex, double distance,
+                                AddressableHeap<Double, ContractionVertex<V>> heap,
+                                Map<ContractionVertex<V>, AddressableHeap.Handle<Double, ContractionVertex<V>>> distanceMap) {
+        AddressableHeap.Handle<Double, ContractionVertex<V>> node = distanceMap.get(vertex);
+        if (node == null) {
+            node = heap.insert(distance, vertex);
+            distanceMap.put(vertex, node);
+        } else if (distance < node.getKey()) {
+            node.decreaseKey(distance);
+        }
+    }
+
+
+    /**
+     * Sets value of {@code isContracted} field of {@code VertexData} for each vertex in the segment
+     * $[independentSetStart,independentSetEnd)$ to $true$. This step should not interfere with other
+     * steps during the contraction because it alters the {@code maskedContractionGraph}.
+     *
+     * @param independentSetStart start of independent set
+     * @param independentSetEnd   end of independent set
+     */
+    private void markContracted(int independentSetStart, int independentSetEnd) {
+        for (int i = independentSetStart; i < independentSetEnd; ++i) {
+            verticesDataArray[((ContractionVertex<V>) verticesArray[i]).vertexId].isContracted = true;
+        }
+    }
+
+
+    /**
+     * Submits {@code tasks} to the {@code completionService} setting start and end
+     * of the working segment and consumer for them
+     *
+     * @param segmentStart start of working segment inclusively
+     * @param segmentEnd   start of working segment exclusively
+     * @param consumer     consumer
+     */
+    private void submitTasks(int segmentStart, int segmentEnd, Consumer<ContractionVertex<V>> consumer) {
```

Comment:
There's another method with the same name, but its logic is a bit different. Let's name them differently

Reply:
This naming was chosen to emphasize the fact that method overloading technique is used in this case.